src/mappings/event/loans.ts

```diff
@@ -0,0 +1,519 @@
+    Transfer,
+    VolumeType,
+} from "../../model";
+import { Ctx, EventItem } from "../../processor";
+import {
+    LoansActivatedMarketEvent,
+    LoansBorrowedEvent,
+    LoansDepositCollateralEvent,
+    LoansDepositedEvent,
+    LoansDistributedBorrowerRewardEvent,
+    LoansDistributedSupplierRewardEvent,
+    LoansNewMarketEvent,
+    LoansRedeemedEvent,
+    LoansRepaidBorrowEvent,
+    LoansUpdatedMarketEvent,
+    LoansWithdrawCollateralEvent
+} from "../../types/events";
+
+import {
+    CurrencyId as CurrencyId_V1020000,
+    CurrencyId_LendToken,
+    Market as LoanMarket_V1020000
+} from "../../types/v1020000";
+
+import { CurrencyId as CurrencyId_V1021000,
+    Market as LoanMarket_V1021000
+} from "../../types/v1021000";
+
+import EntityBuffer from "../utils/entityBuffer";
+import { blockToHeight } from "../utils/heights";
+import { friendlyAmount, getFirstAndLastFour } from "../_utils";
+import { lendTokenDetails } from "../utils/markets";
+
+import { CurrencyId_Token as CurrencyId_Token_V6 } from "../../types/v6";
+import { CurrencyId_Token as CurrencyId_Token_V10 } from "../../types/v10";
+import { CurrencyId_Token as CurrencyId_Token_V15 } from "../../types/v15";
+import { CurrencyId as CurrencyId_V17 } from "../../types/v17";
+import { InterestRateModel as InterestRateModel_V1020000 } from "../../types/v1020000";
+import { address, currencyId, currencyToString, legacyCurrencyId, rateModel } from "../encoding";
+import {
+    updateCumulativeVolumes,
+    updateCumulativeVolumesForCurrencyPair,
+} from "../utils/cumulativeVolumes";
+import { getCurrentRedeemPeriod } from "../utils/requestPeriods";
+import { getVaultId, getVaultIdLegacy } from "../_utils";
+import {Currency} from "../../model/generated/_currency"
+
+
+export async function newMarket(
+    ctx: Ctx,
+    block: SubstrateBlock,
+    item: EventItem,
+    entityBuffer: EntityBuffer
+): Promise<void> {
+    const rawEvent = new LoansNewMarketEvent(ctx, item.event);
+    let e;
+    let underlyingCurrencyId: CurrencyId_V1020000|CurrencyId_V1021000;
+    let market: LoanMarket_V1020000|LoanMarket_V1021000;
+    if (rawEvent.isV1020000) {
+        e = rawEvent.asV1020000;
+        [ underlyingCurrencyId, market ] = e;
+    }
+    else if (rawEvent.isV1021000) {
+        e = rawEvent.asV1021000;
+        underlyingCurrencyId = e.underlyingCurrencyId;
+        market = e.market;
+    }
+    else {
+        ctx.log.warn(`UNKOWN EVENT VERSION: LoansNewMarketEvent`);
+        return;
+    }
+
+    const currency = currencyId.encode(underlyingCurrencyId);
+    const id = currencyToString(currency);
+    const InterestRateModel = rateModel.encode(market.rateModel)
+
+    const height = await blockToHeight(ctx, block.height, "NewMarket");
+    const timestamp = new Date(block.timestamp);
+    const lendTokenIdNo = (market.lendTokenId as CurrencyId_LendToken).value;
+
+    const my_market = new LoanMarket({
+        id: `loanMarket_` + id, //lendTokenId.toString(),
```

Comment:
What's the advantage of having the `loanMarket_` string prefixed? Since this would be a query on the loan markets endpoint, I think it better to remove this. Otherwise, I see that whatever is consuming the data, needs to remove the prefix again.

Reply:
Thanks for spotting this. Agreed.
Can we do this in phase 2? 

I'd prefer to get this merged sooner rather than later so I can start rebasing to include both PR 62 and this one for my ongoing work.

Reply:
I think best to merge this PR but then do for @mafux777 to do a PR to fix the ID (if they agree) before they starting to work on the grafana dashboards as they might break when this is changed later.